src/go/rpk/pkg/cli/cmd/cluster/config/export.go

```diff
@@ -0,0 +1,187 @@
+// Copyright 2021 Vectorized, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package config
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/api/admin"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/config"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/out"
+)
+
+func exportConfig(
+	file *os.File, schema admin.ConfigSchema, config admin.Config, all bool,
+) (err error) {
+	template := make(map[string]interface{})
+
+	// Present properties in alphabetical order, providing some pseudo-grouping based on common prefixes
+	keys := make([]string, 0, len(schema))
+	for k := range schema {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+
+	for _, name := range keys {
+		meta := schema[name]
+		curValue := config[name]
+		template[name] = curValue
+
+		visibility := meta.Visibility
```

Comment:
How about:
```go
if v := meta.Visibility; v == "deprecated" || v == "tunable" && !all {
        continue
}
```
?

As well, visibility is used in the first conditional but not the second -- probably worth using in either both, or neither (or merge the conditionals), like above.

Reply:
Removed the extraneous `visibility` copy.

I've kept these two if statements separate intentionally: it's subjective, I find the code reads cleaner if we list out the conditions separately (it also lets me add comments explaining each `continue` case which I've just done).